From 3d6e168c7ffa8bcebfc2b9965e4af4b86e9384fa Mon Sep 17 00:00:00 2001
From: David Runge <dave@sleepmap.de>
Date: Sat, 15 Jan 2022 13:23:17 +0100
Subject: [PATCH] Add script flag for files containing output sections

src/include/rules.mk:
Extend foreach loop that populates LDS with the linker scriptfile flag
(`-T`), as linking otherwise fails on LDSCRIPTS providing files with
output sections:

```
/usr/bin/ld: warning: guids.lds contains output sections; did you forget -T?
```

Signed-off-by: David Runge <dave@sleepmap.de>
---
 src/include/rules.mk | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/include/rules.mk b/src/include/rules.mk
index f309f863..12986dee 100644
--- a/src/include/rules.mk
+++ b/src/include/rules.mk
@@ -37,7 +37,7 @@ family = $(foreach FAMILY_SUFFIX,$(FAMILY_SUFFIXES),$($(1)_$(FAMILY_SUFFIX)))
 
 %.so :
 	$(CCLD) $(CCLDFLAGS) $(CPPFLAGS) $(SOFLAGS) \
-		$(foreach LDS,$(LDSCRIPTS),$(LD_DASH_T) $(LDS)) \
+		$(foreach LDS,$(LDSCRIPTS),$(LD_DASH_T) -T $(LDS)) \
 		-o $@ $^ $(LDLIBS)
 	ln -vfs $@ $@.1
 
